pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -194,43 +194,54 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	return err
 }
 
+func (c *Reconciler) getPipelineFunc(tr *v1alpha1.PipelineRun) resources.GetPipeline {
+	var gtFunc resources.GetPipeline = func(name string) (v1alpha1.PipelineInterface, error) {
+		p, err := c.pipelineLister.Pipelines(tr.Namespace).Get(name)
+		if err != nil {
+			return nil, err
+		}
+		return p, nil
+	}
+	return gtFunc
+}
+
 func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) error {
 	// We may be reading a version of the object that was stored at an older version
 	// and may not have had all of the assumed default specified.
 	pr.SetDefaults(v1alpha1.WithUpgradeViaDefaulting(ctx))
 
-	p, err := c.pipelineLister.Pipelines(pr.Namespace).Get(pr.Spec.PipelineRef.Name)
+	getPipelineFunc := c.getPipelineFunc(pr)
+	pipelineMeta, pipelineSpec, err := resources.GetPipelineData(pr, getPipelineFunc)
 	if err != nil {
-		// This Run has failed, so we need to mark it as failed and stop reconciling it
+		c.Logger.Errorf("Failed to determine Pipeline spec to use for pipelinerun %s: %v", pr.Name, err)
 		pr.Status.SetCondition(&apis.Condition{
 			Type:   apis.ConditionSucceeded,
 			Status: corev1.ConditionFalse,
 			Reason: ReasonCouldntGetPipeline,
 			Message: fmt.Sprintf("Pipeline %s can't be found:%s",
```

Comment:
I think this message should change now since we're replacing the pipeline name with the pr name. Something like, 

```go
Message: fmt.Sprintf("Error retrieving pipeline for pipelinerun %s: %s", fmt.Sprintf("%s/%s", pr.Namespace, pr.Name), err)
```

Reply:
👍 make sense, updated error message.